src/main/java/org/myrobotlab/service/Py4j.java

```diff
@@ -26,40 +37,79 @@ public Py4j(String n, String id) {
    * start the gateway service listening on port
    */
   public void start() {
-    if (server == null) {
-      server = new GatewayServer(this);
-      server.start();
-      info("server started listening on %s:%d", server.getAddress(), server.getListeningPort());
+    if (gateway == null) {
+      gateway = new GatewayServer(this);
+      gateway.addListener(this);
+      gateway.start();
+      info("server started listening on %s:%d", gateway.getAddress(), gateway.getListeningPort());
+      handler = (Invoker) gateway.getPythonServerEntryPoint(new Class[] { Invoker.class });
     } else {
       log.info("Py4j gateway server already started");
     }
   }
+  
+  public void newScript() {

```

Comment:
As I understand this method will only open a new script.py file if one isn't open already. Wouldn't it be better if we were able to create multiple new scripts, maybe adding an index to the name to differentiate them (script2.py, etc)?

Reply:
Yes, I agree ... this was copied over from the Python/Jython service .. I want them to be in sync.  An interface isn't currently defined, but should be in the future.  but I did add:

```java
  public void newScript(String scriptName) {
  ```
  I'm avoiding file operations though which has been a long thorn in the side ... for a new release Ogre .. because I'm hoping at that time the webgui will be under npm package manager and we can use a file operations library for user interaction to save/open/save as etc